cmd/contour/servecontext.go

```diff
@@ -172,6 +172,41 @@ func newServeContext() *serveContext {
 // TLSConfig holds configuration file TLS configuration details.
 type TLSConfig struct {
 	MinimumProtocolVersion string `yaml:"minimum-protocol-version"`
+
+	// FallbackCertificate defines the namespace/name of the Kubernetes secret to
+	// use as fallback when a non-SNI request is received
+	FallbackCertificate `yaml:"fallback-certificate,omitempty"`
+}
+
+type FallbackCertificate struct {
+	// Name defines the name of the Kubernetes secret containing certificate
+	// to use as fallback when a non-SNI request is received
+	Name string `yaml:"name"`
+
+	// Namespace defines the namespace of the Kubernetes secret containing certificate
+	// to use as fallback when a non-SNI request is received
+	Namespace string `yaml:"namespace"`
+}
+
+func (ctx *serveContext) fallbackCertificate() (*FallbackCertificate, error) {
+	if len(ctx.TLSConfig.FallbackCertificate.Name) == 0 && len(ctx.TLSConfig.FallbackCertificate.Namespace) == 0 {
```

Comment:
The downside of re-using `k8s.FullName` is that it has a default namespace of `default` which doesn't work with fallback well since I don't want a default namespace.

Reply:
Also in servecontext.go this is looking at the struct type `main.FallbackCertificate` since it has the yaml parsing fields on it. I don't think we should bake those into `k8s.FullName`.

Reply:
https://godoc.org/gopkg.in/yaml.v2#Marshal

> Struct fields are only marshalled if they are exported (have an upper case first letter), and are marshalled using the field name lowercased as the default key.

So the yaml package will do what you want without any need for adding yaml tags.

I don't think that the default namespace will be an issue with `k8s.FullName` since you are always requiring a namespace to be specified here. `FullName` only prints "default" when the default (empty) namespace is actually used.

`FullName` is an open struct so you can do the same checks on it that you are doing now. Duplicating the type then means that we need to manually add consistent `String()` formatting, in addition to the overhead of having two types that do the same thing.

Reply:
What made me pause with that struct is that if you don't set a namespace, but call `String()` it uses `default` as it's default value. So if I added an `IsEmpty()` which looked for both name/namespace were empty, then if you called `string()` it would print out a namespace. 

In either case, if we add mintlsprotocol then we'll need a new struct to hold that.